pkg/controller/argocd/rolebinding.go

```diff
@@ -124,19 +126,22 @@ func (r *ReconcileArgoCD) reconcileRoleBinding(name string, rules []v1.PolicyRul
 		Name:     role.Name,
 	}
 
-	if name == applicationController {
-		if err := applyReconcilerHook(cr, roleBinding, ""); err != nil {
-			return err
-		}
-	}
-
 	controllerutil.SetControllerReference(cr, roleBinding, r.scheme)
```

Comment:
A new `roleBinding` only gets created if it does not exist already and `roleRef` has changed.

Reply:
This would add the Controller reference to the `newRoleBinding` object only which gets created in L147, but it makes sense to move it just above the line where the role is actually getting created. 
Done !!